src/Cache/Cache.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Cache;
+
+use InvalidArgumentException;
+use JsonException;
+use LogicException;
+use TwigCsFixer\Ruleset\Ruleset;
+use TwigCsFixer\Sniff\SniffInterface;
+use UnexpectedValueException;
+
+final class Cache implements CacheInterface
+{
+    private SignatureInterface $signature;
+
+    /**
+     * @var array<string, int>
+     */
+    private array $hashes = [];
+
+    public function __construct(SignatureInterface $signature)
+    {
+        $this->signature = $signature;
+    }
+
+    public function getSignature(): SignatureInterface
+    {
+        return $this->signature;
+    }
+
+    public function has(string $file): bool
+    {
+        return \array_key_exists($file, $this->hashes);
+    }
+
+    /**
+     * @throws LogicException
+     */
+    public function get(string $file): int
+    {
+        if (!$this->has($file)) {
+            throw new LogicException('You should call the \'has\' method prior to calling \'get\'');
```

Comment:
I would prefer something like
```suggestion
            throw new LogicException(sprintf('The file %s is not cached', $file));
```